src/App.jsx

```diff
@@ -10,26 +10,91 @@ function App() {
 
   React.useEffect(() => {
     // Side-effect with an empty dependency list
-    const fetchData = () => {
-      return new Promise((resolve, reject) => {
-        setTimeout(() => {
-          const storedData = localStorage.getItem("todoList");
-          const parsedData = storedData ? JSON.parse(storedData) : []; //Added by myself, not sure if it should look like this
-          resolve({ data: { todoList: parsedData } }); // Pass todoList as initial/default list state
-        }, 2000);
-      });
+    const fetchData = async () => {
```

Comment:
In the assignment instructions I think it says to create the async function above the useEffect and just call it in the useEffect. I've had troubles with async functions within useEffect hooks before. I also think it says to have the trycatch after creating the url and options variable, but as long as its working correctly for you then I don't see a big issue with it